lib/console/deployments/pr/impl/bitbucket.ex

```diff
@@ -84,16 +139,66 @@ defmodule Console.Deployments.Pr.Impl.BitBucket do
   defp owner(_), do: nil
 
   defp connection(conn) do
-    with {:ok, url, token} <- url_and_token(conn, "https://api.bitbucket.org/2.0"),
+    with {:ok, url, token} <- url_and_token(conn, @bitbucket_api_url),
       do: Connection.new(url, token)
   end
 
   defp get_pull_id(url) do
-    with {:ok, %URI{path: "/" <> path}} <- URI.parse(url),
-         [org, repo, "pull-requests" <> number] <- String.split(path, "/") do
-      {:ok, "#{org}/#{repo}", number}
+    with %URI{path: "/" <> path} <- URI.parse(url),
+         parts <- String.split(path, "/"),
+         [workspace, repo, "pull-requests", pr_id] <- parts do
+      {:ok, workspace, repo, pr_id}
     else
       _ -> {:error, "could not parse bitbucket url"}
     end
   end
+
+  defp get_pr_info(conn, workspace, repo, pr_id) do
+    with {:ok, pr_body} <- get(conn, "/repositories/#{workspace}/#{repo}/pullrequests/#{pr_id}"),
+         diff_url = pr_body["links"]["diff"]["href"],
```

Comment:
oh wait, these should still be extracted via pattern match instead of nested gets though, since the former is way more robust to errors